runtime/drivers/duckdb/connectors.go

```diff
@@ -19,26 +19,33 @@ func (c *connection) Ingest(ctx context.Context, env *connectors.Env, source *co
 		return err
 	}
 
-	// Driver-specific overrides
-	// switch source.Connector {
-	// case "local_file":
-	// 	return c.ingestFile(ctx, env, source)
-	// }
-
+	// todo :: check if this exceptional handling can be merged
 	if source.Connector == "local_file" {
-		return c.ingestFile(ctx, env, source)
+		return c.ingestLocalFiles(ctx, env, source)
 	}
 
-	path, err := connectors.ConsumeAsFile(ctx, env, source)
+	localPaths, err := connectors.ConsumeAsFile(ctx, env, source)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(path)
+	defer fileutil.RemoveFiles(localPaths)
+	// multiple parquet files can be loaded in single sql
+	// this seems to be performing very fast as compared to appending individual files
+	return c.ingestFiles(ctx, source, localPaths)
+}
 
-	return c.ingestFromRawFile(ctx, source, path)
+// for files downloaded locally from remote sources
+func (c *connection) ingestFiles(ctx context.Context, source *connectors.Source, filenames []string) error {
+	from, err := getSourceReader(filenames)
+	if err != nil {
+		return err
+	}
+	query := fmt.Sprintf("CREATE OR REPLACE TABLE %s AS (SELECT * FROM %s);", source.Name, from)
+	return c.Exec(ctx, &drivers.Statement{Query: query, Priority: 1})
 }
```

Comment:
Actually, we should also pass `CSVDelimiter` to `getSourceReader` here (for remote sources)

Reply:
Yeah, it's probably a mistake. So let's fix it here by passing it into `getSourceReader` and setting it if it's a CSV